### ISSUE TITLE
@MeteC are you interested in snoCAD-X ?

### ISSUE THREAD

**Issue by splitn2** [mikemag/snoCAD-X#6]
@MeteC are you interested in snoCAD-X ? We're looking for help to develop and improve snoCAD-X and make it interoperate with MonkeyCAM, have a look at this and see if it spins your wheels. cheers, Rich

**Comment by bill622** [comment]
Yes, I think it's a great tool. I've added a simple feature: new tip filler shape, commited and did a "pull request" but I never got any feedback.

I have a few ideas of features to add: better camber/rocker adjustments, sidecut radius with bumps like Lib Tech Magnetraction or Arbor, flat kick tip like Capita is doing, splitboard standard insert patterns, etc.

Tell me more about what I could do.

**Comment by splitn2** [comment]
hi @bill622 , its great to know you are interested in contributing and thanks for your input. I know right now @mikemag is flat out busy on other stuff and has not got time right now to focus on the project, but he is definitely keen for experienced enthusiastic developers to be involved. Maybe we can get you and Mike talking, we agree on priorities and specifications/changes, workout some sharing of the coding between you guys, and then do the testing with a wider team incl myself. I wouldn't know where to start with programming nor leading a programming group thats Mikes domain, and I'm assuming you and Mike would need to be on the same thinking and direction and programming strategy to ensure as a team that the outputs are on target and it is a truly cohesive effort - so yeah you guys should talk..

Interesting point on Magnatraction or similar bumps and I'd love to see it in there too at some stage (I'm not sure I want to bend edges to match though!!!), we'd need to understand any patent issues there, so keeping our noses clean on both snowboard design patents and also as we do for the open source software making sure that we are not reusing code that is subject to some other license restriction. Any thoughts on that?

Pretty sure Mike has the splitboard specific code already kind of covered, its just not in MonkeyCAM 4 yet, and it would also need to be in snoCAD-X too, thats one thing I am very very keen on!!

If we can get through the next releases of snoCAD-X and MonkeyCAM to get the current top priority features and tweaks sorted that would be good progress. One thing for sure I am so impressed with the CAM output gcode its a game changer for my future board building. I'm going to show @metec the software tonight as he is visiting me so I will try twist his arm into being involved too!

cheers
Rich

**Comment by mikemag** [comment]
@bill622 I'm glad to hear you're interested in helping out. I'm really sorry for the delay on the pull request... I failed to turn on notifications for this repo. I've fixed that so I'll be more on top of it going forward.

There is a **ton** of stuff we could do to this tool and I'll take any and all help I can get. I've only really skimmed the SnoCAD-X source and ensured it would compile before putting the repo up, but that's as far as I've gone. The major plan I have is to integrate it with MonkeyCAM, which will require a few things that we could use help on for sure. I'd also like to see a bunch of UI improvements, more shape options, etc.

What I want to recommend is this: let's start creating issues for the things we'd like done and we can collaborate on design via the individual issues. From there you can decide what work you want to pick up and run with.

**Comment by splitn2** [comment]
good stuff @bill622 @mikemag let me know how I can help eh, this is an awesome project!

**Comment by danielgraf** [comment]
There is a way already to do magnetraction style serrations in snoCAD, but I'm not sure it's 'correct' - I never built a board using that. If you select the Rail-Shaper from the Tools menu of the board window, and change "Serrations" to a nonzero number - and you can see it really well with 14 for example - then increase the serration depth you see the rails develop 'bumps' along the length. I've never printed one of these or had it cut on a CAM setup so I'm not sure at real-world resolution if the shape is what you'd really want to ride - but there's something there to tweak.

Another possible issue is that the serrations are 'additive' - they extend out from the true curved rail which is ok - until you run out of base material or something on the serrations near the end of the rail.

**Comment by bill622** [comment]
True, I've seen this, but I think the best would be to set a sidecut radius and then add different point where we can pull out of the standard radius. This way it would avoid jerky sidecut curves.

**Comment by splitn2** [comment]
I've invited a couple of kiwi guys to chime in with ideas on magnatraction like edge designs. I agree with Matthew that the wavy serrated style (AKA libtech) is probably more hassle, and simpler "straight lines" pulled out of the radius (like Jones mellow magnatraction ???) is a lot easier to implement and is very effective riding improvement. Have both as options!!!! The Jones style with a straight line of edge interrupting the curve sidecut radius effectively creates nice non catchy pressure points on the edge to get "bite" and I think is best put in both foe and heel areas on each side where the main rider weight and pressure is exerted to the edge. Sound good? @bill622 @mikemag @danielgraf

**Comment by bill622** [comment]
I can't juge which technology is better than others. I think we should stay away from copying patented technology and just offer a way of customizing the standard sidecut radius.

How would you feel about having a standard radius then having the possibility to add control points wherever you want and pull them out to create those magne-traction thing. These control point should also have the hability to adjust the width of the pull out. I hope I'm being clear enough.

Before diving into coding something, I would like to know your ideas about this issue.

**Comment by splitn2** [comment]
@bill622 that sounds good I am pretty sure I understand you. The standard radius sidecut is customized by the user, keep it simple and legit is sensible.

Perhaps allow anything from 2 to 6 of the custom sections to be enabled, and the ability to let the software apportion the gaps between them automatically and evenly, and also alternatively let the user exactly place the custom sections in specific spots.

**Comment by bill622** [comment]
I'll create an issue with this specific feature and assign it to me. I'll check this later on

**Comment by splitn2** [comment]
I can;t assign but @mikemag can?

**Comment by bill622** [comment]
I've just realize I can't. In fact, I can only work on my forked version of SnoCad-X

**Comment by splitn2** [comment]
ahhh right @bill622 I guess if you fork a version and work on it, document what you are doing well enough for Mike to understand it, then provided he likes your work (I think he does already!) he'll merge your code into the main stream. I'm really looking forward to how this all develops ! Hey are you on Facebook? It would be good to be able to message each other to keep in step more frequently if we are bouncing ideas around. cheers Rich

**Comment by danielgraf** [comment]
I made a change in my fork last night to increase control point sizing, then from netbeans I could directly push to a remote upstream, so I picked Mike's master and was able to commit the file there. Maybe Mike gave me permissions to do that though, not sure.

**Comment by danielgraf** [comment]
I'm here on FB : https://www.facebook.com/runzh3

**Comment by splitn2** [comment]
I see you come up as a collaborator so maybe you have full permissions rather than pull/merge style, I'm sure Mike trusts you with snoCAD Dan since you wrote it haha. cheers

**Comment by mikemag** [comment]
@splitn2 is correct: "collaborators" in a repo have full access to it, and can push without any form of review. The usual model with Github is you have a few repo owners, and then everyone else collaborates by forking, making local changes, and putting up pull requests. The pull requests are a great vehicle for code reviews.

**Comment by bill622** [comment]
That's fine with me
